### PR TITLE
Test only: Validate PDB conversion

### PR DESCRIPTION
**Pull request by tmat**
Updates PDB validation methods to perform PDB conversion and validate the converted PDB against the expected value.

### CODE REVIEW

**Comment by tmat** [comment]
@dotnet/roslyn-compiler Please review. Test only change.

**Comment by tmat** [comment]
@ivanbasov Please review.

File: src/Compilers/VisualBasic/Test/Emit/PDB/PDBExternalSourceDirectiveTests.vb [hunk 1]
```diff
@@ -369,7 +369,8 @@ End Class
             </scope>
         </method>
     </methods>
-</symbols>)
+</symbols>, ' Since the CDI Is Not emitted to Portable PDB it won't be present in the converted Windows PDB.
```
Comment (jcouv) [inline src/Compilers/VisualBasic/Test/Emit/PDB/PDBExternalSourceDirectiveTests.vb]: Nit: your comment received VB formatting

File: src/Test/PdbUtilities/Pdb/PdbValidation.cs [hunk 2]
```diff
@@ -192,7 +192,7 @@ public static void VerifyPdb(this CompilationDifference diff, IEnumerable<Method
             IEnumerable<EmbeddedText> embeddedTexts = null,
             IMethodSymbol debugEntryPoint = null,
             DebugInformationFormat format = 0,
-            PdbToXmlOptions options = 0,
+            PdbValidationOptions options = 0,
```
Comment (jcouv) [inline src/Test/PdbUtilities/Pdb/PdbValidation.cs]: Use enum value for default instead?

File: src/Test/PdbUtilities/Pdb/PdbValidation.cs [hunk 3]
```diff
@@ -216,31 +216,139 @@ public static void VerifyPdb(this CompilationDifference diff, IEnumerable<Method
...
+            bool expectedIsXmlLiteral,
+            bool isPortable)
+        {
+            peStream.Position = 0;
+            pdbStream.Position = 0;
+            var actualPdb = XElement.Parse(PdbToXmlConverter.ToXml(pdbStream, peStream, pdbToXmlOptions, methodName: qualifiedMethodName)).ToString();
+            var (actual, expected) = AdjustToPdbFormat(actualPdb, expectedPdb, actualIsPortable: isPortable, actualIsConverted: false);
 
-                AssertEx.AssertLinesEqual(expectedXml.ToString(), actualXml.ToString(), $"PDB format: Portable{Environment.NewLine}", expectedValueSourcePath, expectedValueSourceLine, escapeQuotes: !expectedIsXmlLiteral);
+            AssertEx.AssertLinesEqual(expected, actual, $"PDB format: {(isPortable ? "Portable" : "Windows")}{Environment.NewLine}", expectedValueSourcePath, expectedValueSourceLine, escapeQuotes: !expectedIsXmlLiteral);
```
Comment (jcouv) [inline src/Test/PdbUtilities/Pdb/PdbValidation.cs]: Consider breaking line

File: src/Test/PdbUtilities/Pdb/PdbValidation.cs [hunk 4]
```diff
@@ -216,31 +216,139 @@ public static void VerifyPdb(this CompilationDifference diff, IEnumerable<Method
...
+            foreach (var e in pendingRemoval)
+            {
+                e.Remove();
+            }
+
+            RemoveEmptyScopes(xml);
+            return xml.ToString();
+        }
+
+        internal static (string Actual, string Expected) AdjustToPdbFormat(string actualPdb, string expectedPdb, bool actualIsPortable, bool actualIsConverted)
```
Comment (jcouv) [inline src/Test/PdbUtilities/Pdb/PdbValidation.cs]: @cston will appreciate the CamelCased tuple names ;-)

File: src/Test/PdbUtilities/Pdb/PdbValidation.cs (hunk 2 above)
Comment (tmat) [reply src/Test/PdbUtilities/Pdb/PdbValidation.cs]: Fixed.

File: src/Compilers/VisualBasic/Test/Emit/PDB/PDBExternalSourceDirectiveTests.vb (hunk 1 above)
Comment (tmat) [reply src/Compilers/VisualBasic/Test/Emit/PDB/PDBExternalSourceDirectiveTests.vb]: Fixed.

File: src/Test/PdbUtilities/Pdb/PdbValidation.cs (hunk 3 above)
Comment (tmat) [reply src/Test/PdbUtilities/Pdb/PdbValidation.cs]: Fixed.

**Comment by tmat** [comment]
test ubuntu_16_debug_prtest please

**Comment by tmat** [comment]
test windows_debug_vs-integration_prtest pleas

**Comment by tmat** [comment]
test windows_debug_vs-integration_prtest please

**Pull request merged**